ext/opcache/zend_accelerator_util_funcs.c

```diff
@@ -353,12 +353,31 @@ unsigned int zend_accel_script_checksum(zend_persistent_script *persistent_scrip
 		mem  += (unsigned char*)persistent_script - mem;
 	}
 
+	// zend_class_entry.inheritance_cache is mutated, so we need to reset it before
+	HashTable inheritance_cache_backup;
+	zend_hash_init(&inheritance_cache_backup, zend_hash_num_elements(&persistent_script->script.class_table), NULL, NULL, 0);
+	zend_class_entry *class;
+	ZEND_HASH_FOREACH_PTR(&persistent_script->script.class_table, class) {
+		if (!class->inheritance_cache) {
+			continue;
+		}
+		zend_hash_add_ptr(&inheritance_cache_backup, class->name, class->inheritance_cache);
```

Comment:
It's possible to use plain array instead of HashTable - just store original value of class->inheritance_cache in ``inheritance_cache_backup[i++]``